client/checkout/blocks/upe.js

```diff
@@ -96,10 +96,7 @@ Object.entries( enabledPaymentMethodsConfig ).map( ( [ upeName, upeConfig ] ) =>
 		ariaLabel: __( 'WooCommerce Payments', 'woocommerce-payments' ),
 		supports: {
 			showSavedCards: getUPEConfig( 'isSavedCardsEnabled' ) ?? false,
-			showSaveOption:
-				( getUPEConfig( 'isSavedCardsEnabled' ) &&
-					! getUPEConfig( 'cartContainsSubscription' ) ) ??
-				false,
+			showSaveOption: getUPEConfig( 'showSaveOption' ) ?? false,
```

Comment:
> attempts to reach a top level property named showSaveOption on the woocommerce_payments_data object when that property actually exists a couple levels lower

Oh, I planned to get it from the particular payment method, but I used the wrong place by mistake. 🤕 My bad. Many thanks for noticing this! Fixed.

> I'm wondering if we should just continue handling this logic on the front-end

Regardless of whether we have everything available on the UI or not, I'd prefer to keep any domain logic within the domain layer on the back end. Gateway/checkout classes are the ones in charge of giving us an idea of whether they can be saved.  

Another advantage of keeping the logic on the back end is that all the scripts (e.g. shortcode-classic/shortcode-upe/blocks-classic/blocks-upe) can re-use this flag instead of copying the cascaded if statement from each other and duplicate the code, which is an ideal environment for potential bugs in the future due to 'I updated there but forgot to update here'.

Let me know how it sounds to you and if you agree or you'd like to discuss more. Looking forward to hearing from you! 

> The is_reusable flag

I'll refer to this in another comment.

Reply:
> Another advantage of keeping the logic on the back end is that all the scripts (e.g. shortcode-classic/shortcode-upe/blocks-classic/blocks-upe) can re-use this flag instead of copying the cascaded if statement from each other and duplicate the code, which is an ideal environment for potential bugs in the future due to 'I updated there but forgot to update here'.

Quoting myself here. I've just noticed that we already have a perfect place that has benefited from such an approach - 3a435eb9cb4641a408697a6cad06ecea390445c7.

Reply:
>Regardless of whether we have everything available on the UI or not, I'd prefer to keep any domain logic within the domain layer on the back end. Gateway/checkout classes are the ones in charge of giving us an idea of whether they can be saved.

>Another advantage of keeping the logic on the back end is that all the scripts (e.g. shortcode-classic/shortcode-upe/blocks-classic/blocks-upe) can re-use this flag instead of copying the cascaded if statement from each other and duplicate the code, which is an ideal environment for potential bugs in the future due to 'I updated there but forgot to update here'.

Yes, this all makes perfect sense to me. I'm all for keeping the logic in the back-end for those reasons. Thanks for expanding on the reasons for doing so.